networkx/algorithms/tests/test_efficiency.py

```diff
@@ -1,54 +1,80 @@
 # test_efficiency.py - unit tests for the efficiency module
 #
-# Copyright 2015 NetworkX developers.
+# Copyright 2017 NetworkX developers.
 #
 # This file is part of NetworkX.
 #
 # NetworkX is distributed under a BSD license; see LICENSE.txt for more
 # information.
-"""Unit tests for the :mod:`networkx.algorithms.efficiency` module."""
+# Unit tests for the :mod:`networkx.algorithms.efficiency` module.
+
 from __future__ import division
 from unittest import TestCase
-
 from nose.tools import assert_equal
-
 import networkx as nx
 
 
-def test_efficiency():
-    G = nx.cycle_graph(4)
-    assert_equal(nx.efficiency(G, 0, 1), 1)
-    assert_equal(nx.efficiency(G, 0, 2), 1 / 2)
+class TestEfficiency:
 
+    def __init__(self):
+        # G1 is a toy graph
+        self.G1 = nx.barbell_graph(10, 3)
+        # G2 is a disconnected graph
+        self.G2 = nx.Graph()
+        self.G2.add_nodes_from([1, 2, 3])
+        # G3 is a cycle graph
+        self.G3 = nx.cycle_graph(4)
+        # G4 is a complete graph
+        self.G4 = nx.complete_graph(5)
+        # G5 is the triangle graph with one additional edge
+        self.G4 = nx.lollipop_graph(3, 1)
 
-def test_global_efficiency():
-    G = nx.cycle_graph(4)
-    assert_equal(nx.global_efficiency(G), 5 / 6)
+    def test_efficiency(self):
+        """
+        Returns efficiency of two nodes
+        """
+        return nx.efficiency(self.G1, 1, 10)
 
+    def test_efficiency_disconnected_nodes(self):
+        """
+        Returns 0 when nodes are disconnected
+        """
+        return nx.efficiency(self.G2, 1, 2)
 
-def test_complete_graph_global_efficiency():
-    """Tests that the average global efficiency of the complete graph is
-    one.
+    def test_local_efficiency(self):
+        """
+        Test local efficiency
+        """
+        return nx.local_efficiency(self.G1)
 
-    """
-    for n in range(10):
```

Comment:
hmm... It also look it had a bug, which is why you probably didn't keep the for loop.

Why not put it back, but use:

```
for n in range(2, 10):
    G = nx.complete_graph(n)
    assert_equal(nx.global_efficiency(G), 1)
```